### PR TITLE
Add Decorator Design Pattern

### PR DESCRIPTION
**Pull request by aquaruiz**

### CODE REVIEW

**Comment by aquaruiz** [comment]
@qnislavnachev

File: design-patterns/structural-patterns/src/decorator/Application.java [hunk 1]
```diff
@@ -0,0 +1,16 @@
+package decorator;
+
+public class Application {
+
+  public static void main( String[] args ) {
+	  Sandwich mySandwich = new VegetarianSandwich();
+	  SauceSandwich mySandwichWithSauce = new SauceSandwich(mySandwich);
+	  
+	  SauceSandwich mySandwichWithDoubleSauce = new SauceSandwich(new SauceSandwich(mySandwich));
+	  mySandwichWithDoubleSauce.makeWith("mustard");
```
Comment (qnislavnachev) [inline design-patterns/structural-patterns/src/decorator/Application.java]: Try to add add 1 more sauce to the sandwich. I believe this wont work as expected

**Review by qnislavnachev — CHANGES_REQUESTED** [review]

File: design-patterns/structural-patterns/src/decorator/SauceSandwich.java [hunk 2]
```diff
@@ -0,0 +1,21 @@
+package decorator;
+
+public class SauceSandwich implements Sandwich{
```
Comment (qnislavnachev) [inline design-patterns/structural-patterns/src/decorator/SauceSandwich.java]: It is a good code style if you leave a space between the the name and  '{' in classes, methods and thinks like that

**Review by qnislavnachev — CHANGES_REQUESTED** [review]

File: design-patterns/structural-patterns/src/decorator/Application.java (hunk 1 above)
Comment (aquaruiz) [reply design-patterns/structural-patterns/src/decorator/Application.java]: Fixed.

File: design-patterns/structural-patterns/src/decorator/SauceSandwich.java (hunk 2 above)
Comment (aquaruiz) [reply design-patterns/structural-patterns/src/decorator/SauceSandwich.java]: Fixed.

**Review by qnislavnachev — APPROVED** [review]

**Pull request merged**